core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +111,53 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory in the scm
+     * @param pluginName the name of the plugin
+     * @param scm the valid scm link
+     * @return folderPath from the scm
+     */
+    private static String searchPomFiles(File directory, String pluginName, String scm) {
+        File[] files = directory.listFiles();
+        if (files == null) {
+            return scm;
+        }
+        for (File file : files) {
+            try (Stream<Path> paths = Files.find(file.toPath(), 1, (path, $) ->
```

Comment:
This is not the usage of `Files.find` I had in mind. This function can actually do the recursive discovery of folders, that is what the second argument is for:

`maxDepth - the maximum number of directory levels to search`

You can directly start your function with something like:

`Files.find(directory, SomeMaxValueYouDefine, (path...` and then act on it.

Reply:
And you can also directly search for `pom.xml` files, not directories.

Then you get as a result a list of all found `pom.xml` that you can check with your other function, and return the parent path of whatever matching pom you found.

Reply:
Updated the code